src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -273,12 +273,6 @@ public String buildFullName(BuilderContext context) {
         }
     }
 
-    private static final ThreadLocal<List<Field>> FIELD_LIST = new ThreadLocal<List<Field>>() {
```

Comment:
This list is per-field so you would need 1000 values for a single field to make it grow to a capacity of 1000. I think it is quite uncommon? I am personally more concerned about the opposite case that you suddenly index a large document and memory is never reclaimed since the list is stored in a ThreadLocal?